### PR TITLE
feat: yarn migration

### PR DESCRIPTION
**Pull request by g-cheishvili**
## Related Issue(s)

closes None

## Description

- [x] Replaced mentions of yarn in workflow files
- [x] Replaced npm in ci-scripts
- [x] Replaced cache
- [x] Replaced in angular.json cli config
- [x] Added engines
- [n/a] Added webpack loader for yarn.lock file - not needed anymore
- [x] Added vendor enforced version of yarn
- [x] Blocked install using npm

### CODE REVIEW

File: apps/docs/src/app/documentation/core-helpers/stackblitz/stackblitz-dependencies.ts [hunk 1]
```diff
@@ -38,10 +41,14 @@ export class StackblitzDependencies {
...
 
         [...this._dependencies, ...this._ngDependencies].forEach((dep) => {
-            if (packageLockInfo.dependencies && packageLockInfo.dependencies[dep]) {
-                _dependencies[dep] = packageLockInfo.dependencies[dep].version;
+            if (this._packageLockInfo && this._packageLockInfo[dep]) {
+                if (this._packageLockInfo[dep].length === 1) {
+                    _dependencies[dep] = this._packageLockInfo[dep][0].version;
+                } else {
+                    _dependencies[dep] = packageInfo.dependencies[dep];
+                }
```
Comment (g-cheishvili) [inline apps/docs/src/app/documentation/core-helpers/stackblitz/stackblitz-dependencies.ts:49]: I have no idea if this is correct thing

**Review by platon-rov — COMMENTED** [review]
i heard we have a problems with netlify and pnpm? then we don't have other option that go with yarn.

only one thing i noticed that after `yarn install` lock file got changed so it might be accidentally committed but probably we don't want to.

File: apps/docs/src/app/documentation/core-helpers/stackblitz/stackblitz-dependencies.ts (hunk 1 above)
Comment (g-cheishvili) [reply apps/docs/src/app/documentation/core-helpers/stackblitz/stackblitz-dependencies.ts]: Removed 🤣

**Review by InnaAtanasova — APPROVED** [review]

File: .github/workflows/on-push-or-pull.yml [hunk 2]
```diff
@@ -48,24 +48,18 @@ jobs:
         runs-on: ubuntu-latest
         needs: one_run
         steps:
-            - uses: actions/checkout@v2
-            - uses: actions/cache@v2
-              id: cache
+            - uses: actions/checkout@v3
+            - name: Yarn setup
+              uses: DerYeger/yarn-setup-action@v2.0.0
```
Comment (dmitry-stepanenko) [inline .github/workflows/on-push-or-pull.yml:53]: just curious, what's this and why do we need it?

**Review by N1XUS — DISMISSED** [review]
@g-cheishvili please check failed e2e pipelines

**Comment by N1XUS** [comment]
@g-cheishvili it would be nice if you could also provide metrics from the built time comparing npm & yarn & pnpm for github actions and netlify

File: .github/workflows/on-push-or-pull.yml (hunk 2 above)
Comment (g-cheishvili) [reply .github/workflows/on-push-or-pull.yml:53]: This is for installing and caching yarn deps

File: netlify.toml [hunk 3]
```diff
@@ -7,6 +7,6 @@
 
 [build]
   ignore = 'sh ./ignore.sh'
-  environment = { NODE_VERSION = "14.17.0", PYTHON_VERSION="3.7" }
+  environment = { NODE_VERSION = "16.15.1", PYTHON_VERSION="3.7" }
   publish = "dist/apps/docs"
-  command = "NODE_OPTIONS=--max_old_space_size=8192 npm run build-docs-netlify"
+  command = "NODE_OPTIONS=--max_old_space_size=8192 yarn run build-docs-netlify"
```
[the remark this answers is not in the document]
Comment (g-cheishvili) [reply netlify.toml:12]: Done

**Pull request merged**